packages/shared/src/components/widgets/SocialShare.tsx

```diff
@@ -34,6 +35,7 @@ export const SocialShare = ({
   row,
   onClose,
   parentSelector,
+  shareToSquadState,
```

Comment:
I'd actually think it would be better to send squadToShare and onRequestClose then shareToSquadState. Since state is managed in the higher up component it's better if it's left there, and the component here just uses the value and calls a function no matter what function.

Reply:
We still need to set the value from this component, meaning we would still need the dispatch to be passed along.

Reply:
we can pass it as `onRequestClose` and we define in the other component `onRequestClose={() => setSquadToShare(null)}` so in this component we can just say `onRequestClose={onRequestClose}`

Reply:
We don't need the dispatch for anything else here right? or we do? Maybe I'm just not seeing it in the changed code, so if it's needed just never mind my comment.

Reply:
We need it at line 87:

```
<SquadsToShare onClick={(_, squad) => setSquadToShare(squad)} />
```